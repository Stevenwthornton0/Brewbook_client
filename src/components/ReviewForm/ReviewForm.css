.ReviewForm {
    width: 400px;
    height: 240px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    overflow-wrap: break-word;
}

@media screen and (max-width: 500px) {
    .ReviewForm {
        width: 100%;
    }
}
