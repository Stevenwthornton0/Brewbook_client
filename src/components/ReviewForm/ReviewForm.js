import React, { Component } from 'react';
import BreweryContext from '../../contexts/BreweryContext';
import BreweriesApiService from '../../services/breweries-api-service';

export default class ReviewForm extends Component {
    static contextType = BreweryContext;

    handleSubmit = ev => {
        ev.preventDefault()
        const { brewery } = this.context;
        const { rating, text } = ev.target;
        BreweriesApiService.postReview(brewery.id, rating.value, text.value)
            .then(this.context.addReview)
            .then(() => {
                text.value = ''
            })
            .catch(this.context.setError)
    }

    render() {
        return (
            <div>
                <h2>Post a Review</h2>
                <form
                    className='ReviewForm'
                    onSubmit={this.handleSubmit}
                >
                    <label>Rating</label>
                    <select id='rating' name='rating'>
                        <option value='1'>1</option>
                        <option value='2'>2</option>
                        <option value='3'>3</option>
                        <option value='4'>4</option>
                        <option value='5'>5</option>
                    </select>

                    <label>Review</label>
                    <input 
                        required
                        aria-label='Leave a review...'
                        name='text'
                        id='text'
                        cols='30'
                        rows='3'
                        placeholder='Leave a review..'
                    />

                    <button type='submit'>
                        Post Review
                    </button>

                </form>
            </div>
        )
    }
}