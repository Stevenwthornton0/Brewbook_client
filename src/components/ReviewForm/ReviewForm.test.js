import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import ReviewForm from './ReviewForm';

Enzyme.configure({ adapter: new Adapter() })

describe(`ReviewForm smoke test`, () => {
    it(`renders without crashing`, () => {
        const wrapper = shallow(<ReviewForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})