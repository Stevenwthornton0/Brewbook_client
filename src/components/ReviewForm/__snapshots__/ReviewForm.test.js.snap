// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewForm smoke test renders without crashing 1`] = `
<div>
  <h3>
    Post a Review
  </h3>
  <form
    className="ReviewForm"
    onSubmit={[Function]}
  >
    <label>
      Rating
    </label>
    <select
      className="rating"
      id="rating"
    >
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
      <option
        value="3"
      >
        3
      </option>
      <option
        value="4"
      >
        4
      </option>
      <option
        value="5"
      >
        5
      </option>
    </select>
    <label>
      Review
    </label>
    <Textarea
      aria-label="Leave a review..."
      className="reviewText"
      cols="30"
      id="text"
      placeholder="Leave a review.."
      required={true}
      rows="4"
    />
    <Button
      type="submit"
    >
      Post
    </Button>
  </form>
</div>
`;
